### ISSUE TITLE
Google Analytics Enhanced Ecommerce Tracking Impressions wrong

### ISSUE THREAD

**Issue by fthobe** [bigcommerce/cornerstone#1523]
### Expected behavior
When tracking an impression in Google Analytics if the SKU is empty product ID is submitted, otherwise SKU is submitted as SKU for google analytics.

### Actual behavior
Cornerstone sends always Product ID

### Steps to reproduce behavior
Well, do an order ;)

**Comment by lord2800** [comment]
Hi @fthobe,

Unfortunately it's slightly more complex since this is a problem with the library we're using. We're still discussing internally exactly what we want to do, but the expected behavior you have listed there is exactly what we expect too! I can't promise any timelines, but we definitely want to fix this.

**Comment by fthobe** [comment]
Hi @lord2800 this doesn’t sound as it would be possible to fix it!

**Comment by fthobe** [comment]
There’s probably no way to fix it for us, right? We tried moving around some handlebars but all we get is the ID

**Comment by lord2800** [comment]
Yeah, unfortunately. 😞

**Comment by Tiggerito** [comment]
It looks like Google Ads may new be using those IDs for dynamic remarketing. They have to match with what is used on the pages (including structured data) and what is in the merchant feed. I believe the only way to do that is to use the internal product id.

**Comment by fthobe** [comment]
I agree with @Tiggerito for possible implications, unfortunatly not all of us use BigCommerce for automatisation of SEM and dynamic retargeting. To keep consistency the SKU should be compiled with the SKU and default back to ID if SKU is not present to cover all cases including: 

- third party marketing automatisation decoupled from BC
- automatisation using the tools of bigcommerce

**Comment by fthobe** [comment]
@lord2800 Is the library used for this publicly accessible?

**Comment by fthobe** [comment]
Hi @lord2800, any news on this issue?

**Comment by lord2800** [comment]
No update yet, sorry. 😞

**Comment by Tiggerito** [comment]
> The product ID or SKU (e.g. P67890). *Either this field or name must be set.

https://developers.google.com/analytics/devguides/collection/analyticsjs/enhanced-ecommerce

As SKU is an optional and changeable field, it makes sense to use product id only.

The GA reports use SKU as the column heading, and think this is why people go for SKU. But in reality it is an ID, and should be a unique and trusted identifier for the product.

A client has just contacted us about dynamic remarketing issues with the enhanced ecommerce implementation. They keep getting product ID errors. I've not looked into it, but I suspect this is due to these SKUs randomly being used and Google Ads reading those enhanced ecommerce events.

This may mean that people wanting dynamic remarketing can't use enhanced ecommerce. Unfortunately, if Google Ads is linked to GA it looks like it automatically start pulling in the GA ecommerce events and uses them. If that data is bad, it will ruin all the dynamic remarketing data in Google Ads. It would have to be switched off so that another implementation for dynamic remarketing can be used.

A pity as this could make implementing dynamic remarketing so much easier for store owners. In fact, done correctly and you could say enhanced ecommerce has it implemented out of the box. 

I'm also sure Google Ads will start to use more of the enhanced ecommerce tracking data to further improve smart shopping.

Please make it possible to use it!

**Comment by fthobe** [comment]
@lord2800 Hi Lord, we figured out this bug has some other unintended consequences for us: our trade-in module is effectively broken by not having the SKU at hand anymore. Can you indicate us component responsibile for breaking the SKU on front-end and we will try to fix and make a commit.
